examples/tutorials/api/observation_clustering.py

```diff
@@ -0,0 +1,220 @@
+#
+
+fix, ax = plt.subplots(1, 1, figsize=(7, 5))
+obs_A = Observations([])
+obs_B = Observations([])
+for obs in observations:
+    zenith = obs.get_pointing_altaz(time=obs.tmid).zen.deg
+    if zenith < np.median(obs_zenith):
+        ax.plot(zenith, obs.obs_info["MUONEFF"], "d", color="red")
+        obs_A.append(obs)
+    if zenith > np.median(obs_zenith):
+        ax.plot(zenith, obs.obs_info["MUONEFF"], "o", color="blue")
+        obs_B.append(obs)
+ax.set_ylabel("Muon efficiency")
+ax.set_xlabel("Zenith angle (deg)")
+ax.axvline(np.median(obs_zenith), ls="--", color="black")
+
+
+######################################################################
+# This shows the observations grouped by zenith angle. The diamonds
+# are observations which have a zenith angle less than the median value,
+# whilst the circles are observations above the median.
+#
+# `obs_A` and `obs_B` are both `~gammapy.data.Observations` objects which
+# can be utilised in the usual way to show the various properties of the
+# observations i.e. see the :doc:`/tutorials/data/cta` tutorial.
+#
+
+
+######################################################################
+# Hierarchical clustering of observations
+# ---------------------------------------
+#
+# This method shows how to cluster observations based on their IRF quantities,
+# in this case those that have a similar edisp and psf. The
+# `~gammapy.data.utils.get_irfs_features` is utilised to achieve this. The
+# observations are then clustered based on these criteria using
+# `~gammapy.utils.cluster.hierarchical_clustering`. The idea here is to minimise
+# the variance of both edisp and psf within a specific group to limit the error
+# on the quantity when they are stacked at the dataset level.
+#
+# In this example, the irf features are computed for the `edisp-res` and
+# `psf-radius` at 1 TeV. This is stored as a `astropy.table.table.Table`, as shown below.
+#
+
+source_position = SkyCoord(329.71693826 * u.deg, -30.2255890 * u.deg, frame="icrs")
+names = ["edisp-res", "psf-radius"]
+features_irfs = get_irfs_features(
+    observations, energy_true="1 TeV", position=source_position, names=names
+)
+print(features_irfs)
+
+######################################################################
+# Compute standardized features by removing the mean and scaling to unit
+# variance:
+#
+
+scaled_features_irfs = standard_scaler(features_irfs)
+print(scaled_features_irfs)
+
+
+######################################################################
+# The `~gammapy.utils.cluster.hierarchical_clustering` then clusters
+# this table into `t` groups with a corresponding label for each group.
+# In this case, we choose to cluster the observations into two groups.
+# We can print this table to show the corresponding label which has been
+# added to the previous `feature_irfs` table.
+#
+
+
+features = hierarchical_clustering(scaled_features_irfs, fcluster_kwargs={"t": 2})
```

Comment:
Maybe specify that arguments to `scipy.cluster.hierarchy.fcluster` are passed as a dictionary.